File: packages/types/src/session.ts
export type SessionStatus = "in_progress" | "submitted" | "grading" | "completed" | "failed";

export type SessionAnswerMetadata = {
  questionId: string;
  questionOrder: number;
  durationSeconds: number;
  audioFileName?: string;
  mimeType?: string;
  recordedAt: string;
};

export type ExamSession = {
  id: string;
  examId: string;
  status: SessionStatus;
  resultId?: string;
  startedAt: string;
  currentQuestionOrder: number;
  answers: SessionAnswerMetadata[];
  submittedAt?: string;
};

File: apps/api/src/moldo_api/schemas/session.py
from typing import Literal

from pydantic import BaseModel, ConfigDict


SessionStatus = Literal["in_progress", "submitted", "grading", "completed", "failed"]


class ApiDto(BaseModel):
    model_config = ConfigDict(extra="forbid")


class SessionAnswerMetadata(ApiDto):
    questionId: str
    questionOrder: int
    durationSeconds: int
    audioFileName: str | None = None
    mimeType: str | None = None
    recordedAt: str


class ExamSession(ApiDto):
    id: str
    examId: str
    status: SessionStatus
    resultId: str | None = None
    startedAt: str
    currentQuestionOrder: int
    answers: list[SessionAnswerMetadata]
    submittedAt: str | None = None
